### PR TITLE
[Processing] vector selection tool chapter

### LINKED ISSUE #1483: [FEATURE][processing] New extract by expression algorithm
Original commit: https://github.com/qgis/QGIS/commit/74e64645e410dff89685f029ea0169eb825c7689 by nyalldawson

Filters an input layer by expression

### LINKED ISSUE #1528: [FEATURE][processing] Extract by attribute can extract for null/notnull values
Original commit: https://github.com/qgis/QGIS/commit/24ffa15ecf1aa20ef389fad1f0aaf6f235b712da by nyalldawson

Adds support for filtering where an attribute value is null or not null

### LINKED ISSUE #1946: [FEATURE] Port Extract by Expression to new API, allow saving
non matching features to separate output
Original commit: https://github.com/qgis/QGIS/commit/d8260b8c4788f2a03b9c306db4f7046a27fdd851 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #2075: [FEATURE] Drop processing 'Select by Attribute Sum' algorithm
Original commit: https://github.com/qgis/QGIS/commit/d08398f785b9421d0a8df66d34576cd3097da540 by nyalldawson

Tagged as feature to be included in release notes.

Because:

- The use case for this algorithm is very unclear for users - the name
does not describe what the algorithm does, and there's no help
documentation available for the algorithm either. Given this I suspect
that the algorithm is not being put into use.

- The algorithm needs enhancement to be more useful. There's no logic
in place which dictates how neighbouring features are chosen to
dissolve into the selected feature (it's effectively random - you're
just as likely to get a huge narrow polygon stretching across a map as
you are a nice compact cluster). To be more useful the algorithm would
need logic to either minimise the area of the dissolved feature, or
minimise the total number of dissolved features, or ... ?

### PR DESCRIPTION
**Pull request by ghtmtt**
Description of all the Vector Selection algorithms of Processing.
No pictures added, descriptions are self explanatory (IMHO). 

Fix #2075 fix #1946 fix #1528 fix #1483

### CODE REVIEW

**Comment by ghtmtt** [comment]
I think #2075 could be closed. The `Select by Attribute Sum`  algorithm was in this Processing menu.

File: source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst [hunk 1]
```diff
@@ -98,413 +116,285 @@ Parameters
...
 .......
 
-``Selection`` [vector]
-  <put output description here>
+``Non null geometries`` [vector]
+  Vector layer without NULL geometries
 
-Console usage
-.............
+``Null geometries`` [vector]
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst]: actually it is not an optional choice rather it you can choose to have this output or not. I changed the description of above

File: source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst [hunk 2]
```diff
@@ -98,413 +116,285 @@ Parameters
...
 
-  processing.runalg('qgis:randomextract', input, method, number, output)
-
-See also
-........
+.. _qgis_random_extract_within_subsets:
 
 Random extract within subsets
 -----------------------------
+Takes a vector layer and generates a new one that contains only a subset of the
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst]: No I meant that we should explain that the random selection is made based on the id of the features (according to the discussion, and not their spatial placement).
Actually the issue report is about the `random select within subset` alg but I guess it's the same process for all random thingies...

File: source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst [hunk 3]
```diff
@@ -98,413 +117,278 @@ Parameters
...
+features in the input layer.
 
-Description
-...........
-
-Takes a vector layer and generates a new one that contains only a subset of the features in the input layer.
-The subset is defined randomly, using a percentage or count value to define the total number of features in the subset.
-The percentage/count value is not applied to the whole layer, but instead to each category.
-Categories are defined according to a given attribute, which is also specified as an input parameter for the algorithm.
+The subset is defined randomly, using a percentage or count value to define the
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst]: actually the precision asked above could be there, giving something like : The subset is defined randomly, **based on feature IDs,** using a percentage....

File: source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst [hunk 4]
```diff
@@ -98,413 +116,285 @@ Parameters
...
-...........
+The percentage/count value is not applied to the whole layer, but instead to each
+category.
 
-Takes a vector layer and generates a new one that contains only a subset of the features in the input layer.
-The subset is defined randomly, using a percentage or count value to define the total number of features in the subset.
-The percentage/count value is not applied to the whole layer, but instead to each category.
-Categories are defined according to a given attribute, which is also specified as an input parameter for the algorithm.
+Categories are defined according to a given attribute, which is also specified as
+an input parameter for the algorithm.
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vector_selection_tools.rst]: not convinced by erasing the last part of the sentence and merging with previous paragraph?

**Comment by DelazJ** [comment]
Made some more review, otherwise feel free to merge

**Comment by ghtmtt** [comment]
One step further to have all the docs there! Thanks @DelazJ

**Pull request merged**